src/core/App.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @author Lewy Blue / https://github.com/looeee
+ *
+ * parameters = {
+ *
+ *  canvas: <canvas> DOM element,
+ *
+ *  scene: new THREE.Scene(),
+ *
+ *  camera: new THREE.Camera( ... )
+ *
+ *  renderer: new THREE.WebGLRenderer( ... )
+ *
+ * }
+ */
+
+function App( parameters ) {
+
+	parameters = parameters || {};
+
+	if ( parameters.canvas !== undefined ) {
+
+	  this.canvas = parameters.canvas;
+
+	}	else {
+
+		this.canvas = document.body.appendChild( document.createElement( 'canvas' ) );
+		this.canvas.style.width = this.canvas.style.height = '100%';
+
+	}
+
+	this.scene = ( parameters.scene !== undefined ) ? parameters.scene : new THREE.Scene();
+
+	this.camera = ( parameters.camera !== undefined ) ? parameters.camera : new THREE.PerspectiveCamera( 75, this.canvas.clientWidth / this.canvas.clientHeight, 1, 1000 );
```

Comment:
Then I'd change it to call camera update on init so regardless of the type of camera you pass in it's ready to use when you call `app.renderer.render( app.scene, app.camera );`

See this proof-of-concept

https://github.com/mrdoob/three.js/pull/10687